target/cheri-common/cheri-translate-utils.h

```diff
@@ -121,86 +151,160 @@ generate_ddc_checked_load_ptr(TCGv_cap_checked_ptr checked_addr,
                               DisasContext *ctx, TCGv ddc_offset,
                               target_ulong num_bytes)
 {
-    _generate_ddc_checked_ptr(ctx, TB_FLAG_CHERI_DDC_READABLE, CAP_PERM_LOAD,
-                              checked_addr, ddc_offset, num_bytes);
+    _generate_special_checked_ptr(ctx, TB_FLAG_CHERI_DDC_READABLE,
+                                  CAP_PERM_LOAD, checked_addr, ddc_offset,
+                                  num_bytes, true);
```

Comment:
Hm, there's redundancy between the flags parameter and the use_ddc parameter

Reply:
I guess we could infer `use_ddc` by looking at the permission flags instead. Not sure if that's better.

Reply:
Probably not; if it was just R and W then you could have R, W and DDC bool parameters, but there are more flags than that. We could perhaps assert that use_ddc is consistent with the requested flags, though?

Reply:
Haven't done this in the current version. There are only a few callsites, so I think this should be fine as is.